### ISSUE TITLE
Release with GCC6 fixes?

### ISSUE THREAD

**Issue by verdurin** [Yoshimi/yoshimi#4]
Looks like you've already picked up the GCC6 fixes, @abrolag - I've seen them on Fedora now with the mass rebuild for F24.

Just a note to say that I"m looking forward to a release with them included...

**Comment by verdurin** [comment]
Okay, with your recent commit included, the build gets further, but there are more problems (with GCC6):

```
[ 35%] Building CXX object LV2_Plugin/CMakeFiles/yoshimi_lv2.dir/__/Synth/Envelope.cpp.o
cd /builddir/build/BUILD/yoshimi-1.3.8.2/src/LV2_Plugin && /usr/bin/c++   -DADD_COLOUR=0xdfafbf00 -DBASE_COLOUR=0xbfbfbf00 -DCOMMAND_SIZE=80 -DDEFAULT_AUDIO=jack_audio -DDEFAULT_MIDI=jack_midi -DFADEIN_ADJUSTMENT_SCALE=20 -DFF_MAX_FORMANTS=12 -DFF_MAX_SEQUENCE=8 -DFF_MAX_VOWELS=6 -DHALFPI=1.57079632679f -DJACK_LATENCY -DJACK_SESSION -DLOG_2=0.693147181f -DMAX_AD_HARMONICS=128 -DMAX_ALIENWAH_DELAY=100 -DMAX_BANKS_IN_ROOT=127 -DMAX_BANK_ROOT_DIRS=127 -DMAX_ENVELOPE_POINTS=40 -DMAX_EQ_BANDS=8 -DMAX_FILTER_STAGES=5 -DMAX_PHASER_STAGES=12 -DMAX_PRESETS=1000 -DMAX_SUB_HARMONICS=64 -DMIN_ENVELOPE_DB=-60 -DNUM_INS_EFX=8 -DNUM_KIT_ITEMS=16 -DNUM_MIDI_CHANNELS=16 -DNUM_MIDI_PARTS=64 -DNUM_PART_EFX=3 -DNUM_SYS_EFX=4 -DNUM_VOICES=8 -DPAD_COLOUR=0xcfdfaf00 -DPAD_MAX_SAMPLES=96 -DPI=3.1415926536f -DPOLIPHONY=80 -DSUB_COLOUR=0xafcfdf00 -DTWOPI=6.28318530718f -DVELOCITY_MAX_SCALE=8.0f -DXML_BANK=7 -DXML_CONFIG=6 -DXML_INSTRUMENT=1 -DXML_MICROTONAL=3 -DXML_PARAMETERS=2 -DXML_PRESETS=4 -DXML_STATE=5 -DYOSHIMI=\"yoshimi\" -DYOSHIMI_LV2_PLUGIN=1 -DYOSHI_FIFO_DIR=\"/dev/shm\" -Dyoshimi_lv2_EXPORTS -I/builddir/build/BUILD/yoshimi-1.3.8.2/src/LV2_Plugin -I/usr/include/alsa -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/Fl -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/libdrm -I/builddir/build/BUILD/yoshimi-1.3.8.2/src  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m32 -march=i686 -mtune=atom -fasynchronous-unwind-tables  -fPIC   -D'YOSHIMI_VERSION="1.3.8.2"' -D'BASE_INSTALL_DIR="/usr"' -lasound -ljack -o CMakeFiles/yoshimi_lv2.dir/__/Synth/Envelope.cpp.o -c /builddir/build/BUILD/yoshimi-1.3.8.2/src/Synth/Envelope.cpp
/builddir/build/BUILD/yoshimi-1.3.8.2/src/Params/ControlInterface.cpp: In member function 'void ControlInterface::registerControl(std::__cxx11::string, std::__cxx11::string, YoshimiControlType, float, float, float, float, float*)':
/builddir/build/BUILD/yoshimi-1.3.8.2/src/Params/ControlInterface.cpp:66:67: error: no matching function for call to 'make_pair(std::__cxx11::string&, YoshimiControlParams&)'
     controls.insert(make_pair<string, YoshimiControlParams>(id, cp));
                                                                   ^
In file included from /usr/include/c++/6.0.0/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/6.0.0/bits/char_traits.h:39,
                 from /usr/include/c++/6.0.0/string:40,
                 from /builddir/build/BUILD/yoshimi-1.3.8.2/src/Params/ControlInterface.h:23,
                 from /builddir/build/BUILD/yoshimi-1.3.8.2/src/Params/ControlInterface.cpp:24:
/usr/include/c++/6.0.0/bits/stl_pair.h:406:5: note: candidate: template<class _T1, class _T2> constexpr std::pair<typename std::__decay_and_strip<_Tp>::__type, typename std::__decay_and_strip<_T2>::__type> std::make_pair(_T1&&, _T2&&)
     make_pair(_T1&& __x, _T2&& __y)
     ^~~~~~~~~
/usr/include/c++/6.0.0/bits/stl_pair.h:406:5: note:   template argument deduction/substitution failed:
/builddir/build/BUILD/yoshimi-1.3.8.2/src/Params/ControlInterface.cpp:66:67: note:   cannot convert 'id' (type 'std::__cxx11::string {aka std::__cxx11::basic_string<char>}') to type 'std::__cxx11::basic_string<char>&&'
     controls.insert(make_pair<string, YoshimiControlParams>(id, cp));
                                                                   ^
LV2_Plugin/CMakeFiles/yoshimi_lv2.dir/build.make:985: recipe for target 'LV2_Plugin/CMakeFiles/yoshimi_lv2.dir/__/Params/ControlInterface.cpp.o' failed
make[2]: *** [LV2_Plugin/CMakeFiles/yoshimi_lv2.dir/__/Params/ControlInterface.cpp.o] Error 1
```

**Comment by abrolag** [comment]
Thanks for checking on this. It looks like GCC 6 is *much* more fussy than earlier versions. I'm passing this one on to Andrew as he knows this code.

**Comment by danvd** [comment]
Hi All!

Just pushed a fix 8acbc4c. Hope that this error is fixed now :).

**Comment by abrolag** [comment]
Version !.4.0 onwards is now GCC 'clean' so I'll close this.
